src/main/java/com/sprint/mission/service/jcf/main/JCFChannelService.java

```diff
@@ -0,0 +1,181 @@
+
+import com.sprint.mission.service.MessageService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static com.sprint.mission.entity.main.ChannelType.PRIVATE;
+import static com.sprint.mission.entity.main.ChannelType.PUBLIC;
+
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class JCFChannelService implements ChannelService {
+
+    private final ChannelRepository channelRepository;
+    private final ReadStatusRepository readStatusRepository;
+    private final MessageService messageService;
+    private final UserRepository userRepository;
+    private final ChannelMapper channelMapper;
+    private final MessageRepository messageRepository;
+
+    @Override
+    public Channel createPublicChannel(PublicChannelCreateDTO request) {
+        log.info("createPublicChannel = {}", request);
+        return channelRepository.save(channelMapper.toPublicEntity(request, PUBLIC));
+    }
+
+    @Override
+    public Channel createPrivateChannel(PrivateChannelCreateDTO request) {
+        Channel createdChannel = channelRepository.save(channelMapper.toPrivateEntity(PRIVATE));
+        request.participantIds().stream()
+                .map(userId -> {
+                    User participatingUser = userRepository.findById(userId)
+                            .orElseThrow(() -> new CustomException(ErrorCode.NO_SUCH_USER));
+                    return new ReadStatus(participatingUser, createdChannel, createdChannel.getCreatedAt());
+                })// 나중에
+                .forEach(readStatusRepository::save);
+        return createdChannel;
+    }
+
+
+    /**
+     * [ ] 특정 User가 볼 수 있는 Channel 목록을 조회하도록 조회 조건을 추가하고, 메소드 명을 변경합니다. findAllByUserId [ ] PUBLIC 채널
+     * 목록은 전체 조회합니다. [ ] PRIVATE 채널은 조회한 User가 참여한 채널만 조회합니다.
+     */
+    @Override
+    public Channel findById(UUID channelId) {
+        return channelRepository.findById(channelId)
+                .orElseThrow(() -> new CustomException(ErrorCode.NO_SUCH_CHANNEL));
+    }
+
+    @Override
+    public List<Channel> findAll() {
+        return channelRepository.findAll();
+    }
+
+    //
+    @Override
+    public List<ChannelDto> findAllByUserId(UUID userId) {
+        // 쿼리1
+        List<ReadStatus> readStatusList = readStatusRepository.findAllByUser_Id(userId);
+        // 유저가 참여한 Private 채널 리스트
+        List<Channel> participatingPrivateChannel = readStatusList.stream().map(ReadStatus::getChannel).toList();
+
+        List<ChannelDto> channelDtoList = new ArrayList<>();
+        participatingPrivateChannel.forEach((channel)->{
+            // 채널별 ReadStauts들 가져오기
+            // 쿼리2
+            Instant lastMessageAt = messageRepository.findTop1ByChannel_IdOrderByCreatedAtDesc(channel.getId())
+                    .map(BaseEntity::getCreatedAt)
+                    .orElseGet(null);
```

Comment:
채널에 속한 메시지가 하나도 없으면 NPE가 발생하더라구요.   
아래와 같이 수정이 필요할 것 같아요.

 ```suggestion
            Instant lastMessageAt = Instant lastMessageAt = messageRepository.findTop1ByChannel_IdOrderByCreatedAtDesc(channel.getId())
                    .map(BaseEntity::getCreatedAt)
                    .orElse(null);
```